pydantic_ai_examples/chat_app.py

```diff
@@ -27,8 +35,20 @@
 logfire.configure(send_to_logfire='if-token-present')
 
 agent = Agent('openai:gpt-4o')
+database: Database | None = None
+
+
+@asynccontextmanager
+async def lifespan(_app: fastapi.FastAPI):
+    global database
+
+    async with Database.connect() as db:
+        database = db
+        yield
+        database = None
```

Comment:
This is the recommended way by The FastAPI Expert, FYI.
```suggestion
    async with Database.connect() as db:
        yield {"db": db}
```

You retrieve with `request.state.db`.